scripts/utils.js

```diff
@@ -0,0 +1,55 @@
+/* jshint node: true */
+'use strict';
+
+const spawn = require('cross-spawn');
+
+function dirHasChanges(dir) {
+  return exec('git', ['status', dir, '--porcelain'])
+    .then((result) => {
+      if (!result) {
+        return false;
+      }
+
+      // Untracked files are prefixed with '??'
+      // Modified files are prefixed with 'M'
+      // https://git-scm.com/docs/git-status/1.8.1#_output
+      // https://stackoverflow.com/a/6978402/6178885
+      return (
+        result.trim().indexOf('??') === 0 ||
```

Comment:
Having to call `trim()` twice stood out to me?  Any reason not to save that as a variable?